### PR TITLE
make curl fail loudly for non-2xx downloads

### PR DESCRIPTION
**Pull request by ashi009**
`curl` fails silently if the status code of the response is not 2xx. This CL makes it fail loudly, and break the pipe.

### CODE REVIEW

**Comment by meisterT** [comment]
Let's use the short option form as we do with the other options.

**Comment by ashi009** [comment]
> Let's use the short option form as we do with the other options.

Done.

**Comment by ashi009** [comment]
DO NOT SUBMIT, I'll do another sweep over the codebase.

**Comment by ashi009** [comment]
> DO NOT SUBMIT, I'll do another sweep over the codebase.

PTAL, if this goes too far.

File: scripts/docs/Dockerfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ RUN apt-get -qqy update && \
 
 RUN ln -fs /usr/bin/python3.7 /usr/bin/python
 
-RUN curl -Lo /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.3.0/bazelisk-linux-amd64 && \
+RUN curl -f -Lo /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.3.0/bazelisk-linux-amd64 && \
```
Comment (meisterT) [inline scripts/docs/Dockerfile:28]: in general you can combine these short binary options e.g. `curl -fLo /usr/local/bin/bazel ...`

File: src/test/shell/bazel/remote_helpers.sh [hunk 2]
```diff
@@ -185,7 +185,7 @@ function serve_timeout() {
 # connections, which causes flakes.
 function wait_for_server_startup() {
   touch some-file
-  while ! curl http://localhost:$fileserver_port/some-file > /dev/null; do
+  while ! curl -f http://localhost:$fileserver_port/some-file > /dev/null; do
```
Comment (meisterT) [inline src/test/shell/bazel/remote_helpers.sh:188]: I don't think we want it here.

File: scripts/docs/Dockerfile (hunk 1 above)
Comment (ashi009) [reply scripts/docs/Dockerfile:28]: Done.

File: src/test/shell/bazel/remote_helpers.sh (hunk 2 above)
Comment (ashi009) [reply src/test/shell/bazel/remote_helpers.sh:188]: Done.

**Pull request closed without merging**